File: Moohvie/Models/trophyprogress.swift
import Foundation

struct TrophyMilestone: Identifiable {
    let tier: TrophyTier
    let threshold: Int
    let unlockedDate: Date?

    var id: Int { tier.rawValue }
    var isUnlocked: Bool { unlockedDate != nil }
}

struct TrophyProgress: Identifiable {
    let id: String
    let title: String
    let description: String
    let icon: String
    let count: Int
    let tier: TrophyTier?
    let nextThreshold: Int?
    let milestones: [TrophyMilestone]

    var isUnlocked: Bool { tier != nil }
}

File: android/app/src/main/java/noah/moohvie/models/TrophyProgress.kt
package noah.moohvie.models

data class TrophyMilestone(
    val tier: TrophyTier,
    val threshold: Int,
    /** Epoch millis (équivalent de `Date` côté iOS). */
    val unlockedDate: Long?,
) {
    val isUnlocked: Boolean get() = unlockedDate != null
}

data class TrophyProgress(
    val id: String,
    val title: String,
    val description: String,
    val icon: String,
    val count: Int,
    val tier: TrophyTier?,
    val nextThreshold: Int?,
    val milestones: List<TrophyMilestone>,
) {
    val isUnlocked: Boolean get() = tier != null
}
